addons/addon-base-raas/packages/base-raas-services/lib/alb/alb-service.js

```diff
@@ -0,0 +1,314 @@
+    * @param id
+    * @returns {Promise<>}
+    */
+    async findDeploymentItem({ id }) {
+        const [deploymentStore] = await this.service(['deploymentStoreService']);
+        return await deploymentStore.find({ type: 'account-workspace-details', id });
+    }
+
+    /**
+    * Method to create listener rule. The method creates rule using the ALB SDK client.
+    * Tags are read form the resolvedVars so the billing will happen properly
+    *
+    * @param prefix
+    * @param requestContext
+    * @param resolvedVars
+    * @param targetGroupArn
+    * @returns {Promise<string>}
+    */
+    async createListenerRule(prefix, requestContext, resolvedVars, targetGroupArn) {
+        const deploymentItem = await this.getAlbDetails(requestContext, resolvedVars.projectId);
+        const albRecord = JSON.parse(deploymentItem.value);
+        const listenerArn = albRecord.listenerArn;
+        const priority = albRecord.albDependentWorkspacesCount + 1;
+        const subdomain = this.getHostname(prefix, resolvedVars.envId);
+        const params = {
+            ListenerArn: listenerArn,
+            Priority: priority,
+            Actions: [
+                {
+                    TargetGroupArn: targetGroupArn,
+                    Type: "forward"
+                }
+            ],
+            Conditions: [
+                {
+                    Field: "host-header",
+                    HostHeaderConfig: {
+                        Values: [subdomain]
+                    }
+                },
+                {
+                    Field: "source-ip",
+                    SourceIpConfig: {
+                        Values: [resolvedVars.cidr]
+                    }
+                },
+            ],
+            Tags: resolvedVars.tags
+        }
+        const albClient = await this.getAlbSdk(requestContext, resolvedVars);
+        const response = await albClient.createRule(params).promise();
+        return response.Rules[0].RuleArn;
+    }
+
+    /**
+    * Method to delete listener rule. The method deletes rule using the ALB SDK client.
+    *
+    * @param requestContext
+    * @param resolvedVars
+    * @param ruleArn
+    * @returns {Promise<>}
+    */
+    async deleteListenerRule(requestContext, resolvedVars, ruleArn) {
+        const params = {
+            RuleArn: ruleArn,
+        }
+        const albClient = await this.getAlbSdk(requestContext, resolvedVars);
+        await albClient.deleteRule(params).promise();
+    }
+
+    /**
+    * Method to increase the count of alb dependent workspaces in database
+    *
+    * @param requestContext
+    * @param resolvedVars
+    * @returns {Promise<>}
+    */
+    async increaseAlbDependentWorkspaceCount(requestContext, resolvedVars) {
+        const deploymentItem = await this.getAlbDetails(requestContext, resolvedVars.projectId);
+        var albRecord = JSON.parse(deploymentItem.value);
+        albRecord.albDependentWorkspacesCount = albRecord.albDependentWorkspacesCount + 1;
```

Comment:
Do we need to worry about potentially the field not being correctly incremented in case of parallel requests?